File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Model/src/main/java/ro/model/Persoana.java
package ro.model;

public class Persoana extends Entity<Integer>{
    private String nume;
    private String prenume;

    public Persoana(Integer id, String nume, String prenume) {
        super(id);
        this.nume = nume;
        this.prenume = prenume;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    @Override
    public String toString() {
        return "Persoana{" +
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                '}';
    }

    @Override
    public Integer getId() {
        return super.getId();
    }

    @Override
    public void setId(Integer id) {
        super.setId(id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Persoana persoana = (Persoana) o;

        if (nume != null ? !nume.equals(persoana.nume) : persoana.nume != null) return false;
        return prenume != null ? prenume.equals(persoana.prenume) : persoana.prenume == null;
    }

    @Override
    public int hashCode() {
        int result = nume != null ? nume.hashCode() : 0;
        result = 31 * result + (prenume != null ? prenume.hashCode() : 0);
        return result;
    }
}

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/domain/Persoana.cs
namespace AgentiiZboruriC.domain;

using System;

[Serializable]
public class Persoana : Entity<int>
{
    internal string Nume;
    internal string Prenume;

    public Persoana(int id, string nume, string prenume) : base(id)
    {
        this.Nume = nume;
        this.Prenume = prenume;
    }

    public string GetNume()
    {
        return Nume;
    }

    public void SetNume(string nume)
    {
        this.Nume = nume;
    }

    public string GetPrenume()
    {
        return Prenume;
    }

    public void SetPrenume(string prenume)
    {
        this.Prenume = prenume;
    }

    public override string ToString()
    {
        return $"Persoana{{nume='{Nume}', prenume='{Prenume}'}}";
    }

    public int GetId()
    {
        return base.GetId();
    }

    public void SetId(int id)
    {
        base.SetId(id);
    }

    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;

        Persoana persoana = (Persoana)obj;

        return Nume == persoana.Nume && Prenume == persoana.Prenume;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Nume, Prenume);
    }
}